datafusion/core/src/physical_plan/projection.rs

```diff
@@ -64,6 +68,10 @@ pub struct ProjectionExec {
     columns_map: HashMap<Column, Vec<Column>>,
     /// Execution metrics
     metrics: ExecutionPlanMetricsSet,
+    /// Expressions' normalized orderings (as given by the output ordering API
+    /// and normalized with respect to equivalence classes of input plan). The
+    /// projected expressions are mapped by their indices to this vector.
+    orderings: Vec<Option<PhysicalSortExpr>>,
```

Comment:
You are correct in your last sentence. We calculate the order information for each expression that is projected. Since we use this information when determining ordering equivalences, I thought it would be wise to keep it in the executor's state. I would appreciate it if you have any ideas on how to handle this more effectively.